influxql/functions.go

```diff
@@ -255,6 +266,125 @@ func ReduceMean(values []interface{}) interface{} {
 	return nil
 }
 
+// ReduceMedian computes the median of values
+func ReduceMedian(values []interface{}) interface{} {
+	var data []float64
+	// Collect all the data points
+	for _, value := range values {
+		if value == nil {
+			continue
+		}
+		data = append(data, value.([]float64)...)
+	}
+
+	length := len(data)
+	if length < 2 {
+		if length == 0 {
+			return nil
+		}
+		return data[0]
+	}
+	middle := length / 2
+	var sortedRange []float64
+	if length%2 == 0 {
+		sortedRange = getSortedRange(data, middle-1, 2)
```

Comment:
What am I missing here? Isn't getting the median conceptually simple? Just use the standard library to sort the numbers, and take the one that's "in the middle"? Why the discard* functions?